### PR TITLE
copr: fix get_valid_int_prefix() to be compatible with TiDB(#13045)

### LINKED ISSUE #13045: Confusing behavior in the UPDATE statement
## Bug Report

This is an issue from TiDB: https://github.com/pingcap/tidb/issues/31708.
### What version of TiKV are you using?

v6.2.0
### Steps to reproduce

```
DROP TABLE IF EXISTS t0;
CREATE TABLE t0(c0 INT);
INSERT INTO t0 VALUES(0);
UPDATE t0 SET c0 = 1 WHERE 0^('0.5');
```
### What did you expect?
The column c0 should not be updated, as the result of 0^('0.5') is 0.
### What did happened?
```
mysql> UPDATE t0 SET c0 = 1 WHERE 0^('0.5');
Query OK, 1 row affected (0.01 sec)
Rows matched: 1  Changed: 1  Warnings: 0
```

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What is changed and how it works?

Issue Number: Close #13045

What's Changed: Make `get_valid_int_prefix()` to be compatible with TiDB, i.e. cast function needs to give error/warning when input string is like float('0.3').

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
copr: fix get_valid_int_prefix() to be compatible with TiDB
```

### CODE REVIEW

File: components/tidb_query_datatype/src/codec/convert.rs [hunk 1]
```diff
@@ -855,51 +865,65 @@ pub fn get_valid_int_prefix<'a>(ctx: &mut EvalContext, s: &'a str) -> Result<Cow
...
-                // "1.1." or "1e1.1"
+    get_valid_float_prefix_helper(ctx, s, false)
+}
+
+// As TiDB code(getValidFloatPrefix()), cast expr should not give error/warning when input is
+// empty.
+pub fn get_valid_float_prefix_helper<'a>(
+    ctx: &mut EvalContext,
+    s: &'a str,
+    is_cast_func: bool,
```
Comment (wshwsh12) [inline components/tidb_query_datatype/src/codec/convert.rs:876]: When is_cast_func is ture? I don't find the usage.

Comment (guo-shaoge) [reply components/tidb_query_datatype/src/codec/convert.rs:876]: It should be used in impl_cast.rs:cast_string_as_signed_real(). But it already check whether if input string is empty or not. Also the code path is hard to change.

**Review by wshwsh12 — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]

**Pull request merged**